strawberry_django/auth/utils.py

```diff
@@ -0,0 +1,23 @@
+from strawberry.types import Info
+
+
+def get_current_user(info: Info):
```

Comment:
Still missing the return annotation in here

I think it should be `Optional[AbstractUser]`

Reply:
importing `from django.contrib.auth.models import AbstractUser` or from django.contrib.auth.models import AbstractBaseUser won't work as the django application isn't fully loaded at this point.  

If you have a solution., happy to hear it.

Reply:
import it inside a `if TYPE_CHECKING:`

Reply:
Ruff keeps kicking out TYPE_CHECKING.  

I added:

```python
from typing import Optional, TYPE_CHECKING

from asgiref.sync import sync_to_async
from strawberry.types import Info

if TYPE_CHECKING:
    from django.contrib.auth.models import AbstractBaseUser
```

after committing, it turns into:

```python
from typing import Optional

from asgiref.sync import sync_to_async
from django.contrib.auth.models import AbstractBaseUser
from strawberry.types import Info
```

Thoughts?

Reply:
You probably need to annotate it as a string, like use `"AbstractBaseUser"` instead of `AbstractBaseUser`

That way ruff will allow you to leave it inside the `if TYPE_CHECKING`.

Reply:
Actually, I tried to run it before committing as well - and it just doesn't work. Django will not allow you to load the user-model outside of your Django-project.  Or at the very least before all apps are loaded, which causes similar issues.  

Perhaps you want to give it a shot yourself?  I've put as much time in this particular detail as I can.

Reply:
The solution was doing something like this:

![image](https://github.com/strawberry-graphql/strawberry-graphql-django/assets/134025/20ee7661-3478-46eb-855d-37493efceb39)

But having said that, I fixed it for you and actually used our `UserType`, which is a union of `AbstractBaseUser` and `AnonymousUser`